app/modules/enduser_setup.c

```diff
@@ -675,6 +721,27 @@ static int enduser_setup_http_handle_credentials(char *data, unsigned short data
   }
 
 
+
+  char *config_str = (char *) ((uint32_t)strstr(&(data[6]), "config="));
+  if(config_str != NULL)
+  {
+    int config_field_len = LITLEN("config=");
+    char *config_str_start = config_str + config_field_len;
+    int config_str_len = enduser_setup_srch_str(config_str_start, "& ");
+    /* Create a new variable for storing the decoded string.
+     * Add one more size(char) to allow for \0 */
+    uint8 decoded_config_str[config_str_len+1];
+    ENDUSER_SETUP_DEBUG("Extra Config Data found");
+	
+
+    err |= enduser_setup_http_urldecode(decoded_config_str, config_str_start, config_str_len, sizeof(decoded_config_str));
+	
+    enduser_setup_write_file_with_extra_configuration_data(decoded_config_str, config_str_len);
```

Comment:
The return value of `enduser_setup_write_file_with_extra_configuration_data()` is being ignored here. It probably shouldn't be. Also, that's one long function name. Quite descriptive though, can't argue with that :)